### DISCUSSION TITLE
How to set a certain application as default handler for Publii's preview?

### DISCUSSION THREAD

**Discussion by kevinmatthes** [GetPublii/Publii#1428] [Help]
Hello and thank you for this software!

I just freshly installed Publii v.0.45.1 (build 16608) to the managed Debian 12 VM ChromeOS offers as a developer feature and tried to render and preview an entirely new project in the top-level Chrome browser.  But since I also have VS Code installed, the resulting HTML document is passed to the editor instead of the browser.  Is there any setting for Publii to have it passing the generated preview HTML document to the host OS's Chrome browser?

- host OS:  ChromeOS 121.0.6167.212 official
- VM OS:  Debian 12
- Publii version:  v.0.45.1 (build 16608), Debian 64-bit package from official download section on Publii website
- target browser:  Chrome browser of host OS

**Comment by dziudek** [comment]
Hi,

At this moment it is impossible, we have plan in v.0.47 to address this issue (it is non-trivial to solve).

**Reply by kevinmatthes** [reply]
Thank you for your fast response and the outlook!  I will try to determine some workarounds, until then, and post them here if they succeed in case other users have the same issue.

**Comment by kevinmatthes** [comment]
I found a partial solution:  ChromeOS users need to redirect files of type `text/html` to `garcon_host_browser.desktop` via `~/.local/share/applications/mimeapps.list`.  This will at least render the HTML document but without any CSS.  To reproduce this workaround, users need to add the following lines to their local `mimeapps.list` or adjust existing entries accordingly:

```
[Default Applications]
text/html=garcon_host_browser.desktop
```

ChromeOS uses `garcon_host_browser.desktop` as a synonym for the configured default browser which does not necessarily need to be Chrome but might be any installed Android app, as well.  By overriding the default HTML handler softly using a local configuration file, system definitions of the VM do not need to be altered.  A more persistent approach would be to alter every occurrence in the `/usr` directory as described in https://support.google.com/pixelbook/thread/21786095/launch-chrome-from-linux-beta-app?hl=en which inspired this solution.  Users looking for a possibility to add the host OS's default browser just as another possible application to handle HTML files without assigning any default might be also interested in https://help.gnome.org/admin/system-admin-guide/stable/mime-types-application-user.html.en which can be used as a syntax reference for the configuration file format.

**Reply by kevinmatthes** [reply]
@dziudek

I think I found the reason why CSS is not loaded on preview.

ChromeOS does not access the rendered website directly as the links in the preview files assume (`file:///...`), it creates a copy of the HTML document to preview in a temporary directory which has a random prefix not meeting the path assumptions of the rendered preview.  If the rendered preview could point to the respective destinations using relative paths instead of absolute ones, the links would not break and the site could be shown properly.

Is there an option in Publii to render relative instead of absolute links into the preview?

**Comment by avelguenin** [comment]
Hi !

I'm just writing to say that I have a similar issue on Ubuntu 22.04.5 LTS, using either v.0.39.1 or v.0.46.3. The preview is automatically directed to balenaEtcher, which of course doesn't handle websites.
